File: split-merge/get_x_count.cpp
#include <iostream>
#include <vector>

constexpr int INF = 1e9;

int solve(int n, std::vector<int>& a) {
    std::vector<int> max_pos(n + 1, 0);
    std::vector<int> min_pos(n + 1, INF);
    int val;
    for (int i = 1; i <= n; i++) {
        int val = a[i];
        if (max_pos[val] < i) {
            max_pos[val] = i;
        }
        if (min_pos[val] > i) {
            min_pos[val] = i;
        }
    }
    int pre = 0;
    int ans = 0;
    for (int i = 1; i <= n; i++) {
        if (max_pos[i] == 0) {
            continue;
        }
        if (pre != 0 && min_pos[i] < max_pos[pre]) {
            ans++;
        }
        pre = i;
    }
    return ans;
}

int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);
    int n;
    std::cin >> n;
    std::vector<int> a(n + 1);
    for (int i = 1; i <= n; i++) {
        std::cin >> a[i];
    }
    int ans = solve(n, a);
    std::cout << ans << "\n";
}

File: split-merge/get_x_count.py
import sys
import math

def solve(n, a):
    max_pos = [0] * (n + 1)
    min_pos = [math.inf] * (n + 1)
    for i, val in enumerate(a, start=1):
        if max_pos[val] < i:
            max_pos[val] = i
        if min_pos[val] > i:
            min_pos[val] = i
    ans = 0
    pre = 0
    for i in range(1, n + 1):
        if max_pos[i] == 0:
            continue
        if pre != 0 and min_pos[i] < max_pos[pre]:
            ans += 1
        pre = i
    return ans

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    a = [0] + list(map(int, sys.stdin.readline().split()))
    ans = solve(n, a)
    print(ans)